### PR TITLE
Kinky Dungeon Persistence

### PR DESCRIPTION
**Pull request by Ada18980**
Makes the kinky dungeon minigame persistent so you can leave the game and come back to it without resetting your progress. To start a new game you have to either lose or relog.

### CODE REVIEW

File: BondageClub/Screens/MiniGame/KinkyDungeon/KinkyDungeon.js [hunk 1]
```diff
@@ -33,32 +35,37 @@ function KinkyDungeonDressPlayer() {
...
-		if (group == "ItemM") {
-			if (InventoryGet(Player, "ItemMouth"))
-				KinkyDungeonRestraintsLocked.push("ItemMouth")
-			if (InventoryGet(Player, "ItemMouth2"))
-				KinkyDungeonRestraintsLocked.push("ItemMouth2")
-			if (InventoryGet(Player, "ItemMouth3"))
-				KinkyDungeonRestraintsLocked.push("ItemMouth3")
+	if (KinkyDungeonGameRunning) {
+		// We continue
+	} else {
```
Comment (jomshir98) [inline BondageClub/Screens/MiniGame/KinkyDungeon/KinkyDungeon.js:41]:
```suggestion
	if (!KinkyDungeonGameRunning) {
```

File: BondageClub/Screens/MiniGame/KinkyDungeon/KinkyDungeon.js [hunk 2]
```diff
@@ -33,32 +35,37 @@ function KinkyDungeonDressPlayer() {
...
+		if (!KinkyDungeonPlayer)
+			KinkyDungeonPlayer = CharacterLoadNPC("NPC_Avatar");
+		
+		//KinkyDungeonCreateMap(MiniGameDifficulty);
+		var appearance = CharacterAppearanceStringify(Player)
```
Comment (jomshir98) [inline BondageClub/Screens/MiniGame/KinkyDungeon/KinkyDungeon.js:46]:
```suggestion
		let appearance = CharacterAppearanceStringify(Player);
```

File: BondageClub/Screens/MiniGame/KinkyDungeon/KinkyDungeon.js [hunk 3]
```diff
@@ -33,32 +35,37 @@ function KinkyDungeonDressPlayer() {
...
+		CharacterReleaseTotal(KinkyDungeonPlayer)
+		CharacterNaked(KinkyDungeonPlayer)
+		KinkyDungeonDressPlayer()
+		
+		KinkyDungeonKeybindings = Player.KinkyDungeonKeybindings
+		
+		KinkyDungeonState = "Menu"
+		
+		for (let G = 0; G < KinkyDungeonStruggleGroupsBase.length; G++) {
+			var group = KinkyDungeonStruggleGroupsBase[G]
```
Comment (jomshir98) [inline BondageClub/Screens/MiniGame/KinkyDungeon/KinkyDungeon.js:57]:
```suggestion
			let group = KinkyDungeonStruggleGroupsBase[G];
```

**Comment by Ada18980** [comment]
Thanks @jomshir98

@Ben987 I think this would make a good candidate for a hotfix, given players' complaints about the lack of persistence during the beta and the minimal changes to the code. However it is more of an improvement rather than a bugfix so I understand if not.

**Comment by Ben987** [comment]
Thanks for that fix!  Cheers :)

**Pull request merged**